tests/unit/s2n_session_ticket_test.c

```diff
@@ -1348,6 +1348,54 @@ int main(int argc, char **argv)
         EXPECT_SUCCESS(s2n_config_free(config));
     }
 
+    /* Test: TLS1.3 resumption is successful when key used to encrypt ticket is in decrypt-only state */
+    if (s2n_is_tls13_fully_supported()) {
```

Comment:
Does this test fail if you revert your fix? What is the failure behavior?

Reply:
It will fail with `feature not implemented` without the check

Reply:
I don't mean the s2n_is_tls13_fully_supported() check. I mean that this new test (not just the single line I commented on) is supposed to test the bug that you're fixing. So if you revert your fix, does the test fail, and how does it fail?

Reply:
The test will fail with this change reverted upon calling `s2n_negotiate_test_server_and_client()` with `S2N_ERR_NO_VALID_SIGNATURE_SCHEME`, presumably because it runs into tls1.2 specific path. Current change makes sure we exit [s2n_decrypt_session_ticket()](https://github.com/aws/s2n-tls/blob/b169d76b38061cc4d65aa1236bcce9e404dc8623/tls/s2n_resume.c#L881) early if it is negotiated with tls1.3

Reply:
S2N_ERR_NO_VALID_SIGNATURE_SCHEME isn't an obvious error that you'd expect from incorrectly setting a TLS1.2 handshake type flag on a TLS1.3 connection. In this context, S2N_ERR_NO_VALID_SIGNATURE_SCHEME suggests that the connection fell back to attempt a full handshake, but failed because you haven't configured the connection to be able to do a full handshake (specifically, it's missing a certificate). If you investigate, I suspect you'll find that the key behavior change is that s2n_decrypt_session_ticket() failed before, triggering the connection to reject the session ticket and fall back to a full handshake. After your fix, s2n_decrypt_session_ticket() succeeds, the session ticket is accepted, and a resumption handshake is performed.

I'd suggest taking the time to understand the failure behavior and confirm that it makes sense. Otherwise, you could be missing something important.

Reply:
It might also be useful to set a certificate on your server here so that the test fails when asserting the type of the handshake, not when performing the handshake. It could make the problem more obvious if the test fails again in the future.